src/slot.c

```diff
@@ -225,6 +226,45 @@ gchar *r_slot_get_checksum_data_directory(const RaucSlot *slot, const RaucChecks
 	return g_steal_pointer(&path);
 }
 
+gboolean r_slot_move_checksum_data_directory(const RaucSlot *slot, const gchar *old_digest, const gchar *new_digest, GError **error)
+{
+	GError *ierror = NULL;
+
+	g_return_val_if_fail(slot, FALSE);
+	g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
+
+	if (!slot->data_directory) {
+		return FALSE;
+	}
+
+	g_autofree gchar *old_sub_directory = g_strdup_printf("hash-%s", old_digest);
+	g_autofree gchar *old_path = g_build_filename(slot->data_directory, old_sub_directory, NULL);
+
+	g_autofree gchar *new_sub_directory = g_strdup_printf("hash-%s", new_digest);
+	g_autofree gchar *new_path = g_build_filename(slot->data_directory, new_sub_directory, NULL);
+
+	if (g_file_test(new_path, G_FILE_TEST_EXISTS)) {
+		if (!rm_tree(new_path, &ierror)) {
+			g_propagate_error(error, ierror);
+			return FALSE;
+		}
+	}
+
+	if (rename(old_path, new_path) == -1) {
+		int err = errno;
+		/* Continue if the old path doesn't exist. */
+		if (errno == ENOENT)
```

Comment:
```suggestion
		if (err == ENOENT)
```
Maybe for consistency?